client/state/index.js

```diff
@@ -132,5 +143,75 @@ export function createReduxStore( initialState = {} ) {
 		isBrowser && window.devToolsExtension && window.devToolsExtension()
 	].filter( Boolean );
 
-	return compose( ...enhancers )( createStore )( reducer, initialState );
+	return enhancers;
 }
+
+/**
+ * Named export function for creating a redux store.
+ * This creates a new redux store within the calypsoStore instance.
+ *
+ * @param { object } initialState Redux state with which to initialize (optional).
+ * @param { array } startingReducers Initial list of reducers to start with.
+ * @param { array } enhancers Assembled list of enhancers for the store.
+ * @return { object } The redux store that was created.
+ */
+export function createReduxStore( initialState, startingReducers = initialReducers, enhancers = createEnhancers() ) {
+	// This creates a normal redux store, but it adds our list of original reducers.
+	// This is so we can modify that list of reducers later on and regenerate it.
+	const reduxStore = compose( ...enhancers )( createStore )( combineReducers( startingReducers ), initialState );
+	reduxStore.reducers = startingReducers;
+
+	/**
+	 * Adds a reducer dynamically to the store.
+	 *
+	 * @param { string } name The top-level context name for the reducer.
+	 * @param { function } func The reducer function, in the format of ( state, action ), returning state.
+	 */
+	function addReducer( name, func ) {
+		if ( reduxStore.reducers[ name ] !== undefined ) {
+			throw new Error( `addReducer(): Reducer by name "${ name }" already in use` );
+		}
```

Comment:
I think we could eventually implement a dependency sort of approach to this, where code declares its dependence on a reducer and the reducer is added/removed based on that. But I see that as a layer on top of `addReducer` and `removeReducer` which would be in a future PR. But that would require holding more state as well.